### ISSUE TITLE
Using renderMetaTags in Next.js 13 head file causes an error because of react-intersection-observer

### ISSUE THREAD

**Issue by zakton5** [datocms/react-datocms#72]
In the new the directory structure for Next.js 13 (`/app`), `head.ts` files are used to handle page metadata. However, because these are server side files only, I get the following error when attempting to import and use `renderMetaTags`.

```
Cannot read properties of undefined (reading 'prototype')
    at _inheritsLoose (webpack-internal:///(sc_server)/./node_modules/react-intersection-observer/react-intersection-observer.js:37:51)
    at eval (webpack-internal:///(sc_server)/./node_modules/react-intersection-observer/react-intersection-observer.js:245:5)
    at eval (webpack-internal:///(sc_server)/./node_modules/react-intersection-observer/react-intersection-observer.js:336:2)
    at __webpack_require__ (/.next/server/webpack-runtime.js:33:43)
    at eval (webpack-internal:///(sc_server)/./node_modules/react-datocms/dist/cjs/Image/index.js:44:37)
```

Here is my code (`head.tsx`):

```tsx
import { renderMetaTags } from 'react-datocms';
import { CMSService } from '../services/cms-service';

async function getData(lang: string) {
  const { homePage: page, site } = await CMSService.getHomePage(lang);
  return { seoTags: page.seoTags, site };
}

export default async function Head({ params: { lang } }) {
  const { seoTags, site } = await getData(lang);

  return (
    <>
      <meta name="viewport" content="width=device-width, initial-scale=1" />
      {renderMetaTags([...seoTags, ...site.favicon])}
    </>
  );
}
```

Here is the documentation for the head file in v13: https://beta.nextjs.org/docs/routing/pages-and-layouts#modifying-head

**Comment by sistrall** [comment]
Hello @zakton5,

thanks for reporting this issue. I've been investigating the problem for some hours now, but there's something that's not clear to me about what actually triggers the problem. It seems to happen as soon as `react-intersection-observer` is imported in `react-datocms`.

At the moment, I found a couple of workarounds.

## Importing the helper directly

You could temporarily replace 

```
import { renderMetaTags } from 'react-datocms';
```

with

```
// @ts-ignore
import { renderMetaTags } from "react-datocms/dist/cjs/Seo/renderMetaTags"
```

- That's not a long-term solution: you should Typescript-ignore the line and depend on the `react-datocms` internal directory structure it's not a best practice.

## Mark the head component with `'use client';`

You could add the following declaration on the first line of the component to say to Next.js that some of the required components use client hooks:

```
'use client';
```

For more info, please [see Next.js documentation](https://beta.nextjs.org/docs/rendering/server-and-client-components#convention).

I will keep looking into other possible options.

**Comment by zakton5** [comment]
Thanks @sistrall, I'll attempt the first solution for now as I want the head file to be server side only. It seems that anything imported into a server component eagerly loads all of its other imports. If, for example, it sees a `createContext` _at all_, it will complain, even if you aren't directly using it. 

Maybe a solution could be to separate the exported files into their own namespaces, or at least expose individual exports. So I could do this: 

```
import renderMetaTags from 'react-datocms/renderMetaTags';
```

or 

```
import { renderMetaTags } from 'react-datocms/ssr';
```

**Comment by Diabl0570** [comment]
The same error happens for me when I use <StructuredText>.

The same fix works here as well.

I have the exact same callstack

**Comment by Diabl0570** [comment]
I spoke too soon, I get an hydration error:

```
Unhandled Runtime Error
Error: Hydration failed because the initial UI does not match what was rendered on the server.

See more info here: https://nextjs.org/docs/messages/react-hydration-error
```

**Comment by sistrall** [comment]
Hi @zakton5 and @Diabl0570,

I just released [version 4.0.8](https://www.npmjs.com/package/react-datocms/v/4.0.8) that should fix the problem since it allows import like this:

```typescript
import { renderMetaTags } from "react-datocms/seo"
```

Let me know if that's ok on your side. :-)
